src/Elm/Parser/Expression.elm

```diff
@@ -24,67 +24,71 @@ expression : Parser State (Node Expression)
 expression =
     Pratt.expression
         { oneOf =
-            [ numberExpression
+            [ referenceExpression
                 |> Pratt.literal
-            , referenceExpression
-                |> Pratt.literal
-            , ifBlockExpression
-            , tupledExpression
-            , recordAccessFunctionExpression
-                |> Pratt.literal
-            , negationOperation
-            , letExpression
-            , lambdaExpression
             , literalExpression
                 |> Pratt.literal
-            , charLiteralExpression
+            , numberExpression
                 |> Pratt.literal
-            , recordExpression
+            , tupledExpression
             , glslExpression
                 |> Pratt.literal
             , listExpression
+            , recordExpression
             , caseExpression
+            , lambdaExpression
+            , letExpression
+            , ifBlockExpression
+            , recordAccessFunctionExpression
+                |> Pratt.literal
+            , negationOperation
+            , charLiteralExpression
+                |> Pratt.literal
             ]
+                |> List.map (\parser config -> Layout.positivelyIndented |> Combine.continueWith (parser config))
```

Comment:
This change (3038846) is actually terrible in terms of optimization, it makes the parser quite a bit slower, so I think I will revert it for now.
It makes the parsing more robust, by preventing things like `foo = a\n+b`, which might be worth doing, but it's a but counterproductive in the context of this PR.

Reply:
Alright, sounds good :+1: